src/core/mac/mac_types.hpp

```diff
@@ -539,8 +539,12 @@ class KeyMaterial : public otMacKeyMaterial, public Unequatable<KeyMaterial>
      */
     bool operator==(const KeyMaterial &aOther) const;
 
+#if OPENTHREAD_CONFIG_PLATFORM_KEY_REFERENCES_ENABLE
+
     KeyMaterial(const KeyMaterial &) = delete;
 
+#endif
```

Comment:
Looks good to me. Thanks @sarveshkumarv  👍 

One small suggestion to move this line close to where `operator=()` is defined (which is also under `#if OPENTHREAD_CONFIG_PLATFORM_KEY_REFERENCES_ENABLE`) so it helps make it clear that they are associated and both are done under this config.